### PR TITLE
 Improve markup for definitions and automatic links

### PR DESCRIPTION
**Pull request by dontcallmedom**
Using infrastructure provided by bikeshed

(deleted #390 by mistake)

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Mar 23, 2022, 6:01 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fdontcallmedom%2Fwebtransport%2F2b47c11fdce46c9edee6a323a2e7fba01ff9336f%2Findex.bs&md-warning=not%20ready)

```
<html><body><p><strong>WARNING: </strong>mysqli_connect(): (HY000/2002): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (111)</p></body></html>
```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/webtransport%23391.)._
</details>

### CODE REVIEW

**Comment by jan-ivar** [comment]
Thanks Dom! This LGTM (I didn't review every line, but overall looks like an improvement).

@yutakahirano do you approve of the formatting changes?

**Review by yutakahirano — COMMENTED** [review]
Thanks!

There are some style changes (e.g., `[=[[Readable=]` => `{{[[Readable]]}}`). Can you describe reasons for such style changes for future consistency? Are there any style guides?

File: index.bs [hunk 1]
```diff
@@ -803,31 +769,34 @@ these steps.
...
    received from the server.
    The getter steps for the `incomingBidirectionalStreams` attribute SHALL be:
-     1. Return [=this=]'s [=[[IncomingBidirectionalStreams]]=].
+     1. Return [=this=]'s {{[[IncomingBidirectionalStreams]]}}.
 : <dfn for="WebTransport" attribute>incomingUnidirectionalStreams</dfn>
 :: A {{ReadableStream}} of unidirectional streams, each represented by a
    {{ReceiveStream}}, that have been received from the server.
    The getter steps for `incomingUnidirectionalStreams` are:
-     1. Return [=this=]'s [=[[IncomingUnidirectionalStreams]]=].
+     1. Return [=this=]'s {{[[IncomingUnidirectionalStreams]]}}.
```
[the remark this answers is not in the document]
Comment (dontcallmedom) [reply index.bs:799]:
```suggestion
     1. Return [=this=].{{[[IncomingUnidirectionalStreams]]}}.
```

File: index.bs [hunk 2]
```diff
@@ -315,7 +282,7 @@ interface WebTransportDatagramDuplexStream {
 
 A {{WebTransportDatagramDuplexStream}} object has the following internal slots.
 
-<table class="data" dfn-for="DatagramDuplexStream">
+<table class="data" dfn-for="WebTransportDatagramDuplexStream" dfn-type="attribute">
```
[the remark this answers is not in the document]
Comment (dontcallmedom) [reply index.bs:285]: that's the best (and most often used) option at the moment, see https://github.com/speced/spec-dfn-contract/issues/1 for related discussion

**Comment by dontcallmedom** [comment]
> There are some style changes (e.g., `[=[[Readable=]` => `{{[[Readable]]}}`). Can you describe reasons for such style changes for future consistency? Are there any style guides?

`[= =]` is used to reference generic definitions, when `{{ }}` is used to reference IDL names; although internal slots aren't exactly IDL names, that's how they're handled by most specs nowadays.

**Comment by jan-ivar** [comment]
@dontcallmedom errors are preventing us from merging this. Could you have a look?
```
  Error: Process completed with exit code 2.
  Error: w3c/spec-prod/v2/action.yml (Line: 113, Col: 24):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 113, Col: 24): Error reading JToken from JsonReader. Path '', line 0, position 0.
```

**Comment by dontcallmedom** [comment]
@jan-ivar should be good now

**Pull request merged**